infra/scripts/install.sh

```diff
@@ -19,143 +19,187 @@
+  esac
+  shift
 done
 
-# Check torch version
-if [ "$_TORCH_VER" != "2.5" ] && [ "$_TORCH_VER" != "2.6" ] && [ "$_TORCH_VER" != "nightly" ]; then
-    echo "Invalid torch version '$_TORCH_VER'"
-    echo "(Use --help to see available torch versions)"
-    exit 1
+###############################################################################
+# Detect (and maybe keep) any existing torch installation
+###############################################################################
+python - <<'PY' >/tmp/.tico_torch_ver 2>/dev/null
+import sys, importlib.util, re, json, os
+spec = importlib.util.find_spec("torch")
+if spec is None:
+    sys.exit(0)
+import torch  # noqa: E402
+v = torch.__version__
+# Extract simple major.minor (2.6 etc.)
+m = re.match(r"^(\d+\.\d+)", v)
+print(json.dumps({"full": v, "family": m.group(1) if m else ""}))
+PY
+
+INSTALLED_TORCH_FULL=""
+INSTALLED_TORCH_FAMILY=""
+if [[ -s /tmp/.tico_torch_ver ]]; then
+  INSTALLED_TORCH_FULL=$(jq -r '.full' /tmp/.tico_torch_ver)
+  INSTALLED_TORCH_FAMILY=$(jq -r '.family' /tmp/.tico_torch_ver)
 fi
-# Check if torch version is nightly
+rm -f /tmp/.tico_torch_ver
+
+# Normalise requested spec to family / exact
+REQUEST_IS_NIGHTLY=""
+REQUEST_IS_EXACT=false
 if [[ "$_TORCH_VER" == "nightly" ]]; then
-  _NIGHTLY=true
-fi
-# Check for conflicting options
-if [[ -n "$_USER_SPECIFIED_CUDA" && "$_CPU_ONLY" = true ]]; then
-  echo "[ERROR] Cannot use --cpu_only and --cuda_ver together"
-  exit 1
+  REQUEST_IS_NIGHTLY=true
+elif [[ "$_TORCH_VER" =~ ^[0-9]+\.[0-9]+$ ]]; then
+  : # family only
+elif [[ "$_TORCH_VER" =~ ^[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+  REQUEST_IS_EXACT=true
+else
+  echo "[ERROR] Unsupported --torch_ver value '${_TORCH_VER}'"; exit 1
 fi
 
-# set index-url
-get_index_url_for_cuda_version() {
-  local version="$1"
-  local is_nightly="$2"
-
-  local major=${version%.*}
-  local minor=${version#*.}
-  local cuda_suffix="cu${major}${minor}"
+# Respect pre-installed Torch if allowed
+SKIP_TORCH_INSTALL=false
+if [[ -n "$INSTALLED_TORCH_FULL" ]]; then
+  if [[ " ${SUPPORTED_FAMILIES[*]} " =~ " ${INSTALLED_TORCH_FAMILY} " ]]; then
+    if [[ "$REQUEST_IS_NIGHTLY" = false && "$REQUEST_IS_EXACT" = false ]]; then
+      echo "[INFO] Supported torch ${INSTALLED_TORCH_FULL} already present — keeping it"
+      SKIP_TORCH_INSTALL=true
+      _TORCH_VER="$INSTALLED_TORCH_FAMILY"   # for later requirements file pick
+    else
+      echo "[INFO] '--torch_ver' explicitly requests ${_TORCH_VER}; will override existing ${INSTALLED_TORCH_FULL}"
+    fi
+  else
+    echo "[WARN] Found unsupported torch ${INSTALLED_TORCH_FULL}; will install supported default"
+  fi
+fi
 
-  echo "https://download.pytorch.org/whl${is_nightly:+/nightly}/${cuda_suffix}"
+###############################################################################
+# CUDA index-URL logic
+###############################################################################
+get_index_url() {
```

Comment:
```suggestion
get_index_url_for_cuda_version() {
```
How about keeping the name?